### PR TITLE
Prevent WooPay opt-in checks from running when WooPayments is not available

### LINKED ISSUE #9942: Zero dollar subscriptions fail without an error
### Describe the bug
When the cart contains a zero dollar subscription that will also renewal for free WooPayments gets added as a payment method but is hidden. This makes the checks for WooPay opt-in get triggered but the fields don't get rendered.

WooPayments should not be an available payment method for zero dollar orders with zero dollar renewals

### To Reproduce
- Make sure you're not logged in to WooPay nor the merchant site
- Create a 0 dollar digital subscription
- Add to cart
- Go to the blocks checkout page
- Type all the fields but leave the phone number empty. Don't use autofill as it might fill in a phone number
- Try to place the order
- It will fail without any warning or errors

### Additional context
Slack thread: p1734016378276979-slack-C02CCT3F1QA

### PR DESCRIPTION
**Pull request by leonardola**
Fixes #9942

#### Changes proposed in this Pull Request
This PR adds checks to prevent initializing WooPayments when a cart does not need a payment and to prevent WooPay opt-in checks from running on those cases.

#### Testing instructions
- Make sure you're not logged in to WooPay nor the merchant site
- Create a 0 dollar digital subscription
- Add to cart
- Go to the blocks checkout page
- Type all the fields but leave the phone number empty. Don't use autofill as it might fill in a phone number
- Make sure the order is placed without any problems or errors
- Test other types of zero dollar products
- Test with other payment methods available and non zero dollar products
- Test trial subscription with a non-zero dollar renewal
- Test other scenarios that might fail

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `9943` or branch name `fix/9942-no-cost-subscription` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/9942-no-cost-subscription) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 816c44230346842b50195f69e71ee86d3da9744d
- Build time: 2024-12-18 18:09:27 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Comment by malithsen** [comment]
@leonardola I am not able to replicate the issue on `trunk` for some reason. On both blocks and shortcode checkout, order goes through without any issue.

Do you happen to have a JN site where the issue is reproducible?

**Comment by leonardola** [comment]
@malithsen I forgot an important step on the testing instructions. You need to make sure the phone number input is never filled in to replicate the error. 

In case you still can't reproduce [I set up this site](https://casually-equivalent.jurassic.ninja/) for you to test

File: client/components/woopay/save-user/checkout-page-save-user.js [hunk 1]
```diff
@@ -193,7 +200,11 @@ const CheckoutPageSaveUser = ( { isBlocksCheckout } ) => {
 			return;
 		}
 
-		if ( isSaveDetailsChecked && ! isPhoneValid ) {
+		if (
+			isSaveDetailsChecked &&
```
Comment (malithsen) [inline client/components/woopay/save-user/checkout-page-save-user.js:204]: It seems like `isSaveDetailsChecked` is truthy even when the Save My Info component is not rendered. Making it return `false` when the component is not visible is likely another solution for the issue.

**Review by malithsen — COMMENTED** [review]
Code looks good. However, I am still unable to replicate the issue locally to test out the PR. 

@alefesouza Would you be able to give this one a look.

File: client/components/woopay/save-user/checkout-page-save-user.js (hunk 1 above)
Comment (leonardola) [reply client/components/woopay/save-user/checkout-page-save-user.js:204]: @malithsen do you think it makes sense to change [these lines](https://github.com/Automattic/woocommerce-payments/blob/4b687737240f7fd19e5cb5ef2addf0d6473e835e/client/components/woopay/save-user/checkout-page-save-user.js#L37-L39) to also take into consideration if `isWCPayChosen` ?

Comment (leonardola) [reply client/components/woopay/save-user/checkout-page-save-user.js:204]: I tested this solution but at that point `isWCPayChosen` is set to false which prevents the checkbox from being pre checked for non $0 carts.

**Review by alefesouza — APPROVED** [review]
I could reproduce the error on `develop` and now works as expected

✅ I can place always $0 digital subscriptions orders when not touching the phone input.
✅ I can place orders with WCPay and other payment methods when the cart total is not $0.
✅ I can place trial subscriptions with non $0 renewal.

**Pull request merged**